identity/src/commonMain/kotlin/com/android/identity/util/ByteArrayUtil.kt

```diff
@@ -0,0 +1,279 @@
+    require(value in validRange) { "Value $value is out of Int8 range" }
+    this[offset] = value.toByte()
+}
+
+fun ByteArray.putInt16(offset: Int, value: Int, validRange: IntRange = Short.MIN_VALUE..Short.MAX_VALUE) {
+    require(value in validRange) { "Value $value is out of Int16 range" }
+    this[offset] = ((value shr 8) and 0xFF).toByte()  // Higher byte.
+    this[offset + 1] = (value and 0xFF).toByte() // Lowe byte.
+}
+
+fun ByteArray.putInt16Le(offset: Int, value: Int, validRange: IntRange = Short.MIN_VALUE..Short.MAX_VALUE) {
+    require(value in validRange) { "Value $value is out of Int16 range" }
+    this[offset] = (value and 0xFF).toByte() // Lower byte.
+    this[offset + 1] = ((value shr 8) and 0xFF).toByte()  // Higher byte.
+}
+
+fun ByteArray.putInt32(offset: Int, value: Int, validRange: IntRange = Int.MIN_VALUE..Int.MAX_VALUE) {
+    require(value in validRange) { "Value $value is out of Int32 range" }
+    this[offset] = ((value shr 24) and 0xFF).toByte()
+    this[offset + 1] = ((value shr 16) and 0xFF).toByte()
+    this[offset + 2] = ((value shr 8) and 0xFF).toByte()
+    this[offset + 3] = (value and 0xFF).toByte()
+}
+
+fun ByteArray.putInt32Le(offset: Int, value: Int, validRange: IntRange = Int.MIN_VALUE..Int.MAX_VALUE) {
+    require(value in validRange) { "Value $value is out of Int32 range" }
+    this[offset] = (value and 0xFF).toByte()
+    this[offset + 1] = ((value shr 8) and 0xFF).toByte()
+    this[offset + 2] = ((value shr 16) and 0xFF).toByte()
+    this[offset + 3] = ((value shr 24) and 0xFF).toByte()
+}
+
+fun ByteArray.putInt64(offset: Int, value: Long, validRange: LongRange = Long.MIN_VALUE..Long.MAX_VALUE) {
+    require(value in validRange) { "Value $value is out of Int64 range" }
+    this[offset] = ((value shr 56) and 0xFF).toByte()
+    this[offset + 1] = ((value shr 48) and 0xFF).toByte()
+    this[offset + 2] = ((value shr 40) and 0xFF).toByte()
+    this[offset + 3] = ((value shr 32) and 0xFF).toByte()
+    this[offset + 4] = ((value shr 24) and 0xFF).toByte()
+    this[offset + 5] = ((value shr 16) and 0xFF).toByte()
+    this[offset + 6] = ((value shr 8) and 0xFF).toByte()
+    this[offset + 7] = (value and 0xFF).toByte()
+}
+
+fun ByteArray.putInt64Le(offset: Int, value: Long, validRange: LongRange = Long.MIN_VALUE..Long.MAX_VALUE) {
+    require(value in validRange) { "Value $value is out of Int64 range" }
+    this[offset] = (value and 0xFF).toByte()
+    this[offset + 1] = ((value shr 8) and 0xFF).toByte()
+    this[offset + 2] = ((value shr 16) and 0xFF).toByte()
+    this[offset + 3] = ((value shr 24) and 0xFF).toByte()
+    this[offset + 4] = ((value shr 32) and 0xFF).toByte()
+    this[offset + 5] = ((value shr 40) and 0xFF).toByte()
+    this[offset + 6] = ((value shr 48) and 0xFF).toByte()
+    this[offset + 7] = ((value shr 56) and 0xFF).toByte()
+}
+
+fun ByteArray.putUInt8(offset: Int, value: UInt,
```

Comment:
I think the proper spelling here is `Uint`, not `UInt`. Also everywhere else.

(Yes, we need a separate file to spell out how to do naming/formatting in our source files  - see b/400393701 internally for this)

Reply:
Discussed, decided on UInt to match Kotlin.